packages/service-clients/end-to-end-tests/azure-client/src/test/tree.spec.ts

```diff
@@ -189,5 +173,178 @@ for (const testOpts of testMatrix) {
 			assert.strictEqual(treeData2.root.length, 1);
 			assert.strictEqual(treeData2.root.at(0), "test string 1");
 		});
+
+		if (!isEphemeral) {
+			{
+				class Nicknames extends sf.array("Nicknames", sf.string) {}
+				class UserData extends sf.map("UserData", [sf.string, sf.number, sf.boolean]) {}
+				class User extends sf.object("User", {
+					name: sf.string,
+					nicknames: Nicknames,
+					data: UserData,
+				}) {}
+
+				it("can read and edit data", async () => {
+					const { container } = await client.createContainer(schema, "2");
+					await container.attach();
+					const view = container.initialObjects.tree1.viewWith(
+						new TreeViewConfiguration({ schema: User, enableSchemaValidation: true }),
+					);
+
+					const tags: [string, string | number | boolean][] = [
+						["Age", 32],
+						["Favorite Snack", "Potato Chips"],
+						["Awake", true],
+					];
+
+					view.initialize(
+						new User({
+							name: "Pardes",
+							nicknames: ["Alex", "Duder"],
+							data: new Map(tags),
```

Comment:
We should actually tolerate an iterable of key value pairs here, so thjis should work:

```suggestion
							data: tags,
```

I wouldn't be surprised if that compiles and fails at runtime, I know we have some issues here, but we should probably test it.

Reply:
Good idea. It does compile, but it also fails at runtime as you predicted. These "known issues" - are they tracked?